sources/light_client.move

```diff
@@ -163,10 +191,15 @@ public(package) fun set_block_hash_by_height(lc: &mut LightClient, height: u64,
     df::add(id, height, block_hash);
 }
 
-public(package) fun set_latest_block(lc: &mut LightClient, light_block: LightBlock) {
-    lc.add_light_block(light_block);
-    lc.set_block_hash_by_height(light_block.height(), light_block.header().block_hash()) ;
-    lc.finalized_height = light_block.height();
+/// Appends light block to the current branch and overwrites the current blockchain head.
+/// Must only be called when we know that we extend the current branch or if we control
+/// the rollback.
+public(package) fun append_block(lc: &mut LightClient, light_block: LightBlock) {
+    let head_hash = light_block.header().block_hash();
+    lc.insert_light_block(light_block);
+    lc.set_block_hash_by_height(light_block.height(), head_hash);
+    lc.head_height = light_block.height();
+    lc.head_hash = head_hash;
```

Comment:
We only update head_hash after extend_chain, right?  And we can get head_hash if we know the head height. So head_height is enough I think.

Reply:
it's an optimization, it's used in few other places as well.